thunder/core/update_aliases.py

```diff
@@ -0,0 +1,106 @@
+
+
+def _involves_viewed_args(bsym, viewed):
+    if bsym.sym.id == prims.PrimIDs.RETURN:
+        return False
+    return any(isinstance(p, TensorProxy) and variableify(p) in viewed for p in bsym.flat_proxy_args)
+
+
+def insert_alias_updates(computation_trace: Trace) -> Trace:
+    swap_map = dict()
+    bsyms = []
+
+    # First pass: identify views, their originals, and operands involved in inplace ops
+    view_groups = []
+    inplace_inputs = set()
+    for bsym in computation_trace.bound_symbols:
+        if _is_inplace_op(bsym) or _is_view_creation_op(bsym):
+            # only interested in the input which is modified by the inplace op
+            in_tensor = variableify(bsym.flat_proxy_args[0])
+            out_tensors = set(map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_outs)))
+            if _is_inplace_op(bsym):
+                inplace_inputs.add(in_tensor)
+                out_tensors = set()
+            for group in view_groups:
+                if in_tensor in group:
+                    group.update(out_tensors)
+                    group.add(in_tensor)
+                    break
+            else:
+                view_groups.append(out_tensors.union({in_tensor}))
+
+    # filter out view groups that don't have any tensors involved in inplace ops
+    view_groups = [group for group in view_groups if len(group.intersection(inplace_inputs)) != 0]
+    viewed = set(reduce(set.union, view_groups, set()))
+    encountered = set()
+
+    for bsym in computation_trace.bound_symbols:
+        if _is_inplace_op(bsym) or _is_view_creation_op(bsym) or _involves_viewed_args(bsym, viewed):
+            in_tensors = map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_args))
+            if _is_inplace_op(bsym):
+                in_tensor = list(in_tensors)[0]
+                in_tensors = {in_tensor}
+            else:
+                in_tensors = set(in_tensors)
+            out_tensors = set(map(variableify, filter(lambda p: isinstance(p, TensorProxy), bsym.flat_proxy_outs)))
+            encountered.update(in_tensors)
+            group = set(reduce(set.union, filter(lambda g: any(g.intersection(in_tensors)), view_groups), set()))
+            if len(group) == 0:
+                # This is a view creation with operands that are not involved in any inplace ops.
+                bsyms.append(bsym.from_bsym_swap_proxies(swap_map, skip_output=True))
+                continue
+            views_encountered = group.intersection(encountered)
+            new_aliases = _get_new_aliases(views_encountered, computation_trace)
+
+            update_bsym, swap_map = _get_update_bsym(views_encountered, swap_map, new_aliases)
+            bsyms.append(update_bsym)
+            encountered.update(out_tensors)
+            new_bsym = bsym.from_bsym_swap_proxies(swap_map)
+            bsyms.append(new_bsym)
+            if _is_inplace_op(bsym) and len(out_tensors) == 1:
```

Comment:
Combined with the previous suggestion about `bsym.flat_proxy_args`.
```suggestion
            if _is_inplace_op(bsym) and len(out_tensors) == 1  and in_tensors:
```